pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -408,13 +408,12 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 // predicate. With this topologyPairsMaps available, the affinity predicate does not
 // need to check all the pods in the cluster.
 func getTPMapMatchingIncomingAffinityAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*schedulercache.NodeInfo) (topologyPairsAffinityPodsMaps *topologyPairsMaps, topologyPairsAntiAffinityPodsMaps *topologyPairsMaps, err error) {
-	allNodeNames := make([]string, 0, len(nodeInfoMap))
-
 	affinity := pod.Spec.Affinity
 	if affinity == nil || (affinity.PodAffinity == nil && affinity.PodAntiAffinity == nil) {
 		return newTopologyPairsMaps(), newTopologyPairsMaps(), nil
 	}
 
+	allNodeNames := make([]string, 0, len(nodeInfoMap))
```

Comment:
I **_thought_** following logic would be more performant:

```go
	allNodeNames := make([]string, len(nodeInfoMap))
	for i, name := range nodeInfoMap {
		allNodeNames[i] = name
	}
```

So I did a [test](https://play.golang.org/p/33bc1GOa1KV), but it turns it's even a little bit less performant...

So current logic is good to me. Thanks!

Reply:
Thanks for checking. The biggest issue here is that in our scalability tests with 5000 nodes, these "allNodeNames" slices which exist in `getTPMapMatchingIncomingAffinityAntiAffinity`, `getTPMapMatchingExistingAntiAffinity`, and `CalculateInterPodAffinityPriority` allocate/release a tone of memory (about 12GB) over the course of the test. The one here is doing it unnecessarily for pods that don't have any affinity rules. The other two are harder to avoid.

Reply:
For getTPMapMatchingExistingAntiAffinity, it seems the assumption is that the number of entries of allNodeNames would be the same as the number of entries of nodeInfoMap when processNode runs.

Is it possible to skip allNodeNames altogether ?
e.g. 
```
  processNode := func(i int) {
    nodeInfo := nodeInfoMap[allNodeNames[i]]
```
Instead of getting nodeInfo from nodeInfoMap by node name, can an iterator of nodeInfoMap be used to skip the first i-1 entries ?
That way, allNodeNames wouldn't be needed.

Reply:
@tedyu I exactly had the same question when I realized that these arrays which are used purely for iterating over nodeInfoMap allocate and release so much memory. I did a bit of looking, but I realized that iterating over maps in Go is not predictable. So, my finding is that it is not possible to go to the "i"th entry of a map.